sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala

```diff
@@ -324,6 +349,41 @@ object WatermarkSupport {
       }
     Some(evictionExpression)
   }
+
+  /**
+   * Find the column which is marked as "event time" column.
+   *
+   * If there are multiple event time columns in given column list, the behavior depends on the
+   * parameter `useFirstOccurrence`. If it's set to true, the first occurred column will be
+   * returned. If not, this method will throw an AnalysisException as it is not allowed to have
+   * multiple event time columns.
+   */
+  def findEventTimeColumn(
+      attrs: Seq[Attribute],
+      useFirstOccurrence: Boolean): Option[Attribute] = {
```

Comment:
The name for this variable is kind of confusing.  Can we rename it something like "allowMultipleEventTimeColumns"?

Reply:
Good suggestion.